src/main/java/com/redhat/devtools/lsp4ij/LanguageServiceAccessor.java

```diff
@@ -91,6 +93,37 @@ private LanguageServiceAccessor(Project project) {
 
     private final Set<LanguageServerWrapper> startedServers = new HashSet<>();
 
+    /**
+     * Check each project for open files and start an LS if matching one is found and is not started yet
+     */
+    public static void checkCurrentlyOpenFiles() {

```

Comment:
no you don't want to restart LS that were manually stopped on purpose. You only need to start the newly added LS if, there are matching opened files

Reply:
Added an explicit check to only connect, if the ls definition matches